### PR TITLE
Add support for version-less APIs

### PR DESCRIPTION
**Pull request by L8D**
So this is the path of least resistance to implementing this feature. Hope this helps!

### CODE REVIEW

File: rest-core/src/Rest/Driver/Routing/Internal.hs [hunk 1]
```diff
@@ -80,9 +80,12 @@ route mtd uri = routeWith defaultConfig mtd uri
...
-routeWith cfg (Just mtd) uri api = runRouter cfg mtd uri $
-  do versionStr <- popSegment
-     case versionStr `Rest.lookupVersion` api of
+routeWith cfg (Just mtd) uri api = runRouter cfg mtd uri $ do
+  case api of
+    Rest.Unversioned (Some1 router) -> routeRoot router
+    Rest.Versioned   vrs            -> do
+      versionStr <- popSegment
+      case versionStr `Rest.lookupVersion` vrs of
           Just (Some1 router) -> routeRoot router
```
Comment (L8D) [inline rest-core/src/Rest/Driver/Routing/Internal.hs]: Whoops. Messed up the indentation a little there.

**Comment by hesselink** [comment]
Just a small note to let you know that I'll review this as soon as possible, it's not forgotten!

**Comment by L8D** [comment]
bump @hesselink

**Comment by hesselink** [comment]
Oh, I'm very sorry I forgot about this :(

I've just looked at it, and it looks very good. The only thing I'm a bit unsure about is `withVersion`. Having it like this means that for unversioned APIs, you'll still need to pass 'latest'. In rest-gen this is the default, so it's no problem. But in the docs handler that means you'll have to have 'latest' in your urls. Is that what you intended? The alternative is for `withVersion` to take a `VersionSet`. That way users will have to unpack the API first, and only call it for a versioned API. What do you think?

**Comment by L8D** [comment]
I think we can leave it with requiring 'latest' for docs. The un-versioned use-case seems to be a small minority and we can deal with whatever inconveniences come with.

**Comment by hesselink** [comment]
Sounds good. I'll merge and test with our app, and then I'll probably release soon. It's a breaking change, but we have no other breaking changes coming up so there's no use in delaying.

**Comment by hesselink** [comment]
I've released rest-core 0.39 with this change, and updated the other libraries to support this version.

Thanks!

**Pull request merged**